### ISSUE TITLE
Cannot store numpy array in Python 3.5, produces TypeError 

### ISSUE THREAD

**Issue by SmokinCaterpillar** [manahl/arctic#142]
Hi the following short snippet fails in Python 3.5:
```

from arctic import Arctic
import numpy as np

mongo_host = 'localhost'
store = Arctic(mongo_host)
store.initialize_library('type_error')

library = store['type_error']

library.write('test', np.array([[1,2],[3,42]]))
```

With the following error trace:
```
Traceback (most recent call last):
  File "/home/robert/PycharmProjects/DEAP_n_pypet/mongotest_minimal2.py", line 10, in <module>
    library.write('test', np.array([[1,2],[3,42]]))
  File "/home/robert/Programme/anaconda/envs/py35/lib/python3.5/site-packages/arctic/decorators.py", line 50, in f_retry
    return f(*args, **kwargs)
  File "/home/robert/Programme/anaconda/envs/py35/lib/python3.5/site-packages/arctic/store/version_store.py", line 562, in write
    mongo_retry(handler.write)(self._arctic_lib, version, symbol, data, previous_version, **kwargs)
  File "/home/robert/Programme/anaconda/envs/py35/lib/python3.5/site-packages/arctic/decorators.py", line 50, in f_retry
    return f(*args, **kwargs)
  File "/home/robert/Programme/anaconda/envs/py35/lib/python3.5/site-packages/arctic/store/_ndarray_store.py", line 387, in write
    self._do_write(collection, version, symbol, item, previous_version)
  File "/home/robert/Programme/anaconda/envs/py35/lib/python3.5/site-packages/arctic/store/_ndarray_store.py", line 400, in _do_write
    projection={'sha': 1, '_id': 0},
TypeError: unhashable type: 'Binary'
```

It works under Python 2.7, though.

**Comment by SmokinCaterpillar** [comment]
Update: It also works under Python 3.4, only fails under 3.5.

**Comment by bmoscon** [comment]
I have a fix for this, once the builds pass and reviewers sign off I'll merge it in

**Comment by bmoscon** [comment]
Actually, this looks like a bug/issue in Pymongo. I've opened a defect with the mongodb team:

https://jira.mongodb.org/browse/PYTHON-1095


If its hashable in 2.7 and 3.4, it should be hashable in 3.5

**Comment by bmoscon** [comment]
Actually, they fixed it in 3.2.2. Please update your pymongo to 3.2.2.

**Comment by bmoscon** [comment]
FYI this was the commit that resolved the issue:

https://github.com/mongodb/mongo-python-driver/commit/3321b66d3404e5b6f92dcd5ef0da599bbb21fbf7

long story short, in Python3 if you override eq, you better override hash too :)
